### PR TITLE
CON-234: fixing link to Python API reference

### PR DESCRIPTION
**Pull request by rkorte**
According to Alex in HMAINT-276, I can count on the Python API URL being like the other API URLs. So although this link to the Python API reference doesn't work yet, it will:
https://community.rti.com/static/documentation/connext-dds/current/doc/api/connext_dds/api_python/intro.html

### CODE REVIEW

**Comment by rkorte** [comment]
I'm keeping 'current' in the URL (rather than using '6.1.1') in case the Python API reference is updated out of band with Connector or Connext DDS releases. Maybe 'current' is the best bet for now.

**Review by samuelraeburn — APPROVED** [review]
Needs to be cherry-picked too

File: docs/intro.rst [hunk 1]
```diff
@@ -11,7 +11,7 @@ leading ultra-high performance, distributed networking databus.
 
 *RTI Connector* is a family of simplified APIs for publishing and subscribing
 to the *Connext DDS* Databus in programming languages such as Python and JavaScript. 
-(RTI also offers a more comprehensive `Python API <https://community.rti.com/static/documentation/connext-dds/current/api/connext_dds/api_python/intro.html>`__, 
+(RTI also offers a more comprehensive `Python API <https://community.rti.com/static/documentation/connext-dds/current/doc/api/connext_dds/api_python/intro.html>`__,
```
Comment (alexcamposruiz) [inline docs/intro.rst:14]: I think we should link to the GitHub repo instead https://github.com/rticommunity/connextdds-py

The documentation is currently built on a best-effort basis, and doesn't necessarily follow our release timeline.

**Review by alexcamposruiz — CHANGES_REQUESTED** [review]

File: docs/intro.rst (hunk 1 above)
Comment (rkorte) [reply docs/intro.rst]: I changed it. 

Once you're at https://github.com/rticommunity/connextdds-py, the link to the Python API Reference goes to a 6.0.1 URL: https://community.rti.com/static/documentation/connext-dds/6.0.1/api/connext_dds/api_python/index.html. I'm assuming that's OK.

**Review by alexcamposruiz — APPROVED** [review]

**Pull request merged**